lib/control-app.js

```diff
@@ -136,5 +135,15 @@ module.exports = function(config) {
         });
     });
 
-    return app;
+  var server = app.listen(0, function () {
+    var port = server.address().port
+    function close (cb) {
+      server.close(function (err) {
+        build.close()
+        clientBundler.close()
+        cb(err)
+      })
+    }
+    cb(null, { port, close })
+  })
```

Comment:
I moved the `.listen()` part from `lib/airtap.js` and callback with the `port` and a `close` method that in turn closes the server and the `watchify` instance.